backend/msgpack/conn.go

```diff
@@ -130,13 +130,8 @@ func (c *connection) Close() {
 	c.Unlock()
 }
 
-func (c *connection) resetAndConnectWithLock() error {
-	c.numFailures = 0
-	c.lastConnectNanos = c.nowFn().UnixNano()
-	return c.connectWithLockFn()
-}
-
 func (c *connection) connectWithLock() error {
+	c.lastConnectNanos = c.nowFn().UnixNano()
```

Comment:
nit: perhaps we should rename this to `lastConnectAttemptNanos`?

Reply:
Sounds good.